app/scripts/controllers/transactions/index.js

```diff
@@ -331,6 +331,195 @@ export default class TransactionController extends EventEmitter {
     });
   }
 
+  // ====================================================================================================================================================
+
+  /**
+   * @param {number} txId
+   * @returns {TransactionMeta} the txMeta who matches the given id if none found
+   * for the network returns undefined
+   */
+  _getTransaction(txId) {
+    const { transactions } = this.store.getState();
+    return transactions[txId];
+  }
+
+  _checkIfTxStatusIsUnapproved(txId) {
+    return (
+      this._getTransaction(txId).status === TRANSACTION_STATUSES.UNAPPROVED
+    );
+  }
+
+  _updateTransaction(txId, txParams, note) {
+    const txMeta = this._getTransaction(txId);
+    const updatedMeta = { ...txMeta, ...txParams };
```

Comment:
`txParams` is a property of `txMeta`. `{ ...txMeta, ...txParams }` will cause the properties of `txParams` to be flattened into `txMeta.
If you just want to update txParams, then this would work: `{ ...txMeta, txParams }` (without the elippses before txParams, so that it's properties are not spread)

Reply:
oh, but I see that the other methods are not just passing what we call `txParams`, but whole objects that can be spread properly

Reply:
In that case, I just suggest renaming `txParams` to `newTxMeta` so that there is not confusion with the `txMeta.txParams` property.